xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -238,9 +238,10 @@ STDMETHODIMP_(ULONG) CDXVADecoderWrapper::AddRef(void)
 STDMETHODIMP_(ULONG) CDXVADecoderWrapper::Release(void)
 {
   AtomicDecrement(&m_refs);
-  assert(m_refs >= 0);
-  if (m_refs == 0) delete this;
-  return m_refs;
+  volatile long refs = m_refs;
```

Comment:
This isn't thread safe, what if someone takes a reference after this line, we'll delete the object and go boom